text/0000-variadic.md

```diff
@@ -102,10 +102,11 @@ consist of aliases for the underlying Rust types, making it unnecessary for
 `libc` to provide additional implementations of the `VaArg` trait. Nothing
 outside of `core` should define any implementation of `VaArg`.
 
-Note that extracting an argument from a `VaList` follows the platform-specific
-rules for argument passing and promotion. In particular, many platforms promote
-any argument smaller than a C `int` to an `int`. On such platforms, extracting
-the corresponding type will extract an `int` and convert appropriately.
+Note that extracting an argument from a `VaList` follows the C rules for
+argument passing and promotion. In particular, C code will promote any argument
+smaller than a C `int` to an `int`, and promote `float` to `double`. Thus,
+Rust's argument extractions for the corresponding types will extract an `int`
+or `double` as appropriate, and convert appropriately.
```

Comment:
Currently, when passing arguments to external variadic functions, Rust disallows `f32`, `i8`, `i16`, `u8`, `u16` and `bool` ([E0617](https://doc.rust-lang.org/error-index.html#E0617)). Should `VaArg` also follow this rule instead of performing promotion implicitly?

Reply:
C does this kind of promotion already; it seems less error-prone to allow Rust code to extract the corresponding type of argument (e.g. `args.next::<u16>()`) rather than extracting a `c_int` and casting.